### ISSUE TITLE
.env files being ignored by watcher

### ISSUE THREAD

**Issue by Arcanorum** [remy/nodemon#1806]
- `nodemon -v`: 2.0.4
- `node -v`: 12.19.0
- Operating system/terminal environment: Ubuntu
- Using Docker? What image: No
- Command you ran: nodemon index.js

### Expected behaviour
That is I change any of the settings in my .env file, that the nodemon will restart, like it does for other files that are being watched.

### Actual behaviour
Nodemon keeps running like nothing happened.

### Steps to reproduce
Create a .env file in the watched directory, run nodemon, modify and save .env.

**Comment by remy** [comment]
And *how* are you telling nodemon to watch this file? Since it's watching js based files when you don't provide any options.

**Comment by Isaius** [comment]
You can create a file nodemon.json and tell to nodemon to watch the extesions you want.

```json
{
  "watch": [".env", "*.js"]
}
```

**Comment by remy** [comment]
As per @Isaius' solution.

**Comment by TobiasHennig** [comment]
Hi @remy, i have a similar problem in my project. The `.env` file does not trigger a restart with the following `nodemon.json`:
`
{ "ext": "env,js,json" }
`

**Comment by remy** [comment]
@TobiasHennig there's multiple reasons for this and potential problems that actually make sense that it *doesn't* trigger a restart.

Firstly a dotenv file is hidden and has no extension. By default, IIRC, chokidar (the underlying file watcher) ignores hidden files (which technically you can modify in the nodemon config). However, the dotenv has no extension, so your watch list doesn't actually track it (there's other issues about tracking extensionless, as yet, years after being filed, it's not been solved by anyone).

The added problem is the expectation that changing a dotenv will change the environment values the process is exposed to. This isn't quite as simple and obvious as you might think.

Nodemon launches processes in two ways. If it's a node script with no special launch conditions, then nodemon will fork. This means the environment values are the ones that nodemon has, which means restarting won't change the environment.

If however it's launched using the spawn method, then the environment is rebuilt, so it would work, but my bet is that most users aren't aware of when each method is being used (and rightful so).

So, in short, env reloading is a problem with multiple vectors. I believe the *right* way to solve this is through process.

Of course, if this can be solved with great UX I'd be more than happy to take a PR, but the reality is that there's very few PRs that come from the public and it tends to rely entirely on me - which is fine, but I prioritise as I feel appropriate.

**Comment by TobiasHennig** [comment]
Hi @remy thank you very much for your time to answer my question. I have started to think about the reason why i want to restart on `.env` change and think it's okay how it works at the moment. Thanks for making and maintaining "nodemon", it saves a lot of  time.

**Comment by ajzkk** [comment]
This work for me `nodemon.json`

```
{
  "watch": [".env", "."],
  "ignore": [
    "**/node_modules/**",
    "**/bower_components/**",
    "**/dist/**"
  ],
  "ext": "ts,pug",
  "exec": "yarn start"
}
```
